src/os/bluestore/BitmapFreelistManager.cc

```diff
@@ -221,6 +221,9 @@ bool BitmapFreelistManager::enumerate_next(uint64_t *offset, uint64_t *length)
 {
   std::lock_guard<std::mutex> l(lock);
 
+  // Don't store this in the object, iterators should be short lived.
+  KeyValueDB::Iterator enumerate_p;
```

Comment:
I don't think this is quite sufficient.. we have to maintain a position across calls, and I don't see where we create the new iterator on each call.  probably teh get_iterator has to move up here, and a lower_bound call to position it so that all paths end up with a valid iterator.